dds_cli/user.py

```diff
@@ -145,22 +147,29 @@ def read_token(self):
 
         if not self.file_exists():
             LOG.debug(f"Token file {self.token_file} does not exist.")
-            return None
+            return None, None
 
         self.check_token_file_permissions()
 
-        if self.token_expired():
-            LOG.debug("No token retrieved from file, will fetch new token from api")
-            return None
-
         # Read token from file
         with self.token_file.open() as file:
             token = file.read()
             if not token:
                 raise exceptions.TokenNotFoundError(message="Token file is empty.")
 
+            # Use lifetime from token header if given, else read default from config
+        try:
+            lft = jwt.get_unverified_header(token).get("lft")
+            lifetime = isodate.parse_duration(lft) if lft else dds_cli.TOKEN_MAX_AGE
+        except:
```

Comment:
It's a general recommendation to avoid `except` clauses like this. The recommendation is to specify the exception you except to be raised instead. The major benefit I can see is that it's much easier to refactor it in the future.

Reply:
Thanks for the helpful advice! Then I guess I will add the base exception [`jwt.exceptions.InvalidTokenError`](https://pyjwt.readthedocs.io/en/latest/api.html#exceptions) or should I be super specific and really add the relevant subclasses?